service/src/main/java/com/zebrunner/reporting/service/reporting/TestSessionServiceV1.java

```diff
@@ -0,0 +1,85 @@
+package com.zebrunner.reporting.service.reporting;
+
+import com.zebrunner.reporting.domain.db.reporting.TestSession;
+import com.zebrunner.reporting.persistence.dao.mysql.application.TestSessionMapper;
+import com.zebrunner.reporting.service.TestService;
+import com.zebrunner.reporting.service.exception.IllegalOperationException;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static com.zebrunner.reporting.service.exception.IllegalOperationException.IllegalOperationErrorDetail.TEST_SESSION_CAN_NOT_BE_UPDATED;
+
+@Service
+public class TestSessionServiceV1 {
+
+    private static final String ERR_MSG_TEST_SESSION_NOT_EXIST_BY_ID = "Test session with id %d cannot be found";
+
+    private final TestSessionMapper testSessionMapper;
+    private final TestService testService;
+
+    public TestSessionServiceV1(TestSessionMapper testSessionMapper, TestService testService) {
+        this.testSessionMapper = testSessionMapper;
+        this.testService = testService;
+    }
+
+    @Transactional
+    public TestSession create(TestSession session) {
+        removeNotExistingTestRefs(session.getTestRefs());
+        testSessionMapper.create(session);
+        testSessionMapper.linkToTests(session.getId(), session.getTestRefs());
+        return session;
+    }
+
+    @Transactional(readOnly = true)
+    public TestSession retrieveById(Long id) {
+        TestSession session = testSessionMapper.findById(id);
```

Comment:
Do you really need `@Transactional` for single db query?

Reply:
I`m using transaction to mark session as read only